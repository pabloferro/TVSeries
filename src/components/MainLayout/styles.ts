import {StyleSheet} from 'react-native';
import {background} from '../../constants/colors';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: background,
  },
  normalContainer: {
    paddingVertical: 16,
    paddingHorizontal: 16,
  },
});
