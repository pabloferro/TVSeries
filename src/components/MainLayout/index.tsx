import React, {ReactNode} from 'react';
import {SafeAreaView} from 'react-native-safe-area-context';
import CustomText from '../CustomText';
import {TextVariants} from '../CustomText/TextVariants';

import styles from './styles';

interface Props {
  children: ReactNode;
  title: string;
}

export default function MainLayout({children, title}: Props) {
  return (
    <SafeAreaView style={styles.container}>
      <CustomText variant={TextVariants.h1} style={styles.title}>
        {title}
      </CustomText>
      {children}
    </SafeAreaView>
  );
}
