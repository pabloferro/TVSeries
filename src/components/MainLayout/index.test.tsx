import React from 'react';
import {render} from '@testing-library/react-native';

import MainLayout from '.';

test('renders correctly', () => {
  const {toJSON} = render(<MainLayout>Children</MainLayout>);
  expect(toJSON()).toMatchSnapshot();
});

test('renders correctly for lists', () => {
  const {toJSON} = render(<MainLayout list>Children</MainLayout>);
  expect(toJSON()).toMatchSnapshot();
});
