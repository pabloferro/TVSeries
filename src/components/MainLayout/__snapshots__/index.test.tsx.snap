// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Array [
      Object {
        "backgroundColor": "#111",
        "flex": 1,
      },
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 16,
      },
      undefined,
    ]
  }
>
  Children
</RNCSafeAreaView>
`;

exports[`renders correctly for lists 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Array [
      Object {
        "backgroundColor": "#111",
        "flex": 1,
      },
      false,
      undefined,
    ]
  }
>
  Children
</RNCSafeAreaView>
`;
